test/main/assert.js

```diff
@@ -300,6 +315,34 @@ QUnit.test( "throws", function( assert ) {
 		/description/,
 		"throw error from property of 'this' context"
 	);
+
+	var errorObjectLiteral = {
+		toString: function() {
+			return "[object";
+		},
+		name: { // bad name
+			toString: function() { }
+		},
+		message: { // bad message
+			toString: function() { }
+		}
+	};
+	assert.throws(
+		function() {
+			throw errorObjectLiteral;
```

Comment:
Interseting.. It seems like it should behave the same as the plain `{ name: "SomeName", message: "some message" }` case earlier. Which branches does this cover?

Reply:
This hits later branches within the "errorString" helper, where name and message can each have falsey values:

```
function errorString( error ) {
	const resultErrorString = error.toString();
 
	// If the error wasn't a subclass of Error but something like
	// an object literal with name and message properties...
	if ( resultErrorString.substring( 0, 7 ) === "[object" ) {
		const name = error.name ? error.name.toString() : "Error";
		const message = error.message ? error.message.toString() : "";
 
		if ( name && message ) {
			return `${name}: ${message}`;
		} else Eif ( name ) {
			return name;
		} else if ( message ) {
			return message;
		} else {
			return "Error";
		}
	} else {
		return resultErrorString;
	}
}
```

I wasn't convinced that `(new CustomError()).toString()` lead down the `[object` path, but I see now that it does. I can make this testpoint simpler and more direct...